packages/vscode-extension/src/handlers.ts

```diff
@@ -624,6 +624,83 @@ export async function addCapabilityHandler(args?: any[]): Promise<Result<null, F
   return result;
 }
 
+export async function addFeatureHandler(args?: any[]): Promise<Result<null, FxError>> {
+  // TODO: move the question model to fx-core addFeature API
+  const addFeatureConfig: SingleSelectConfig = {
+    name: "addFeature",
+    title: "Add features",
+    step: 1,
+    options: [
+      {
+        id: "capability",
+        label: `${localize("teamstoolkit.handlers.addFeature.capability.label")}`,
+        detail: localize("teamstoolkit.handlers.addFeature.capability.detail"),
+      },
+      {
+        id: "resource",
+        label: `${localize("teamstoolkit.handlers.addFeature.resource.label")}`,
+        detail: localize("teamstoolkit.handlers.addFeature.resource.detail"),
+      },
+      {
+        id: "additional",
+        label: `${localize("teamstoolkit.handlers.addFeature.additional.label")}`,
+        detail: localize("teamstoolkit.handlers.addFeature.additional.detail"),
+      },
+    ],
+  };
+  const additionalFeatureConfig: SingleSelectConfig = {
+    name: "additionalFeature",
+    title: "Additional features",
+    step: 2,
+    options: [
+      {
+        id: "sso",
+        label: `${localize("teamstoolkit.handlers.addFeature.sso.label")}`,
+        detail: localize("teamstoolkit.handlers.addFeature.sso.detail"),
+      },
+      {
+        id: "api",
+        label: `${localize("teamstoolkit.handlers.addFeature.api.label")}`,
+        detail: localize("teamstoolkit.handlers.addFeature.api.detail"),
+      },
+      {
+        id: "cicd",
+        label: `${localize("teamstoolkit.handlers.addFeature.cicd.label")}`,
+        detail: localize("teamstoolkit.handlers.addFeature.cicd.detail"),
+      },
+    ],
+  };
+  const stack = [addFeatureConfig];
+  while (stack.length > 0) {
```

Comment:
Why not ask the questions in fx-core and leverage existing question model(maybe move these questions to getQuestionsForAddFeature())? You can also benefit from question model for other platforms.

Reply:
Yes, this logic will be moved to fx-core later. I will change to invoking user task after it's ready.